Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm

```diff
@@ -300,31 +300,40 @@ - (void)menuDidClose:(NSMenu *)menu
     if (!page || !page->preferences().imageAnalysisMarkupEnabled())
         return;
 
-    auto* imageBitmap = m_context.controlledImage();
-    if (!imageBitmap)
+    auto context = m_context.controlledImageElementContext();
+    if (!context)
         return;
 
-    auto image = imageBitmap->makeCGImage();
-    if (!image)
-        return;
+    page->shouldAllowImageMarkup(*context, [protectedThis = Ref { *this }, weakMenu = WeakObjCPtr<NSMenu> { m_menu.get() }](bool shouldAllow) mutable {
+        if (!shouldAllow)
+            return;
 
-    requestImageAnalysisMarkup(image.get(), [protectedThis = Ref { *this }, weakMenu = WeakObjCPtr<NSMenu> { m_menu.get() }](CGImageRef result, CGRect) {
-        if (!result)
+        auto* imageBitmap = protectedThis->m_context.controlledImage();
+        if (!imageBitmap)
             return;
 
-        auto strongMenu = weakMenu.get();
-        if (!strongMenu)
+        auto image = imageBitmap->makeCGImage();
+        if (!image)
             return;
 
-        auto markupImageItem = adoptNS([[NSMenuItem alloc] initWithTitle:contextMenuItemTitleMarkupImage() action:@selector(markupImage) keyEquivalent:@""]);
-        [markupImageItem setImage:[NSImage imageWithSystemSymbolName:@"person.fill.viewfinder" accessibilityDescription:contextMenuItemTitleMarkupImage()]];
-        [markupImageItem setTarget:WKSharingServicePickerDelegate.sharedSharingServicePickerDelegate];
-        [markupImageItem setAction:@selector(markupImage)];
-        if (auto numberOfItems = [strongMenu numberOfItems])
-            [markupImageItem setIndentationLevel:[strongMenu itemAtIndex:numberOfItems - 1].indentationLevel];
-        [strongMenu addItem:markupImageItem.get()];
+        requestImageAnalysisMarkup(image.get(), [protectedThis = WTFMove(protectedThis), weakMenu = WTFMove(weakMenu)](CGImageRef result, CGRect) {
+            if (!result)
+                return;
 
-        protectedThis->m_croppedImageResult = result;
+            auto strongMenu = weakMenu.get();
+            if (!strongMenu)
+                return;
+
+            auto markupImageItem = adoptNS([[NSMenuItem alloc] initWithTitle:contextMenuItemTitleMarkupImage() action:@selector(markupImage) keyEquivalent:@""]);
+            [markupImageItem setImage:[NSImage imageWithSystemSymbolName:@"person.fill.viewfinder" accessibilityDescription:contextMenuItemTitleMarkupImage()]];
+            [markupImageItem setTarget:WKSharingServicePickerDelegate.sharedSharingServicePickerDelegate];
+            [markupImageItem setAction:@selector(markupImage)];
+            if (auto numberOfItems = [strongMenu numberOfItems])
+                [markupImageItem setIndentationLevel:[strongMenu itemAtIndex:numberOfItems - 1].indentationLevel];
```

Comment:
This could be rewritten as `[markupImageItem setIndentationLevel:[strongMenu itemArray].lastObject. indentationLevel];`, since the default `indentationLevel` is 0.

Reply:
Good point! Will change to `[markupImageItem setIndentationLevel:[strongMenu itemArray].lastObject. indentationLevel];`